asmcomp/cmmgen.ml

```diff
@@ -1346,26 +1355,23 @@ let make_switch arg cases actions dbg =
   let is_const = function
     | Cconst_int n
     | Cconst_pointer n -> (n land 1) = 1
-    | Cconst_natint _
-    | Cconst_float _
+    | Cconst_natint n
+    | Cconst_natpointer n -> (Nativeint.(to_int (logand n one) = 1))
```

Comment:
If the low bit is clear, then the array is not a valid OCaml value. I suppose it doesn't really need to be: I think we can be sure that the GC never touches it, since the address of the array never escapes.

Would it be a good idea to allow the low bit to be zero and delete the block header? Then it would be a plain lookup table, not pretending to be an OCaml heap value.